packages/react-native/src/immutable-form/actions.js

```diff
@@ -38,8 +38,14 @@ export const submitForm = (formName: string) => (submit: (Object) => Promise<any
     const state = getState();
 
     submit(state).then(
-      res => dispatch(approvedForm(formName)(res)),
-      err => dispatch(errorForm(formName)(errorMessage(err))),
+      res => {
+        dispatch(clearForm(formName)());
+        dispatch(approvedForm(formName)(res));
```

Comment:
Maybe reverse the order?
First dispatch that the form was approved, and then clear it.
Otherwise, I think that users that depends on the Redux action of `FORM_APPROVED` won't be able to get the field's values